src/request-pipeline/index.js

```diff
@@ -257,16 +266,12 @@ function isDestResBodyMalformed (ctx) {
     return !ctx.destResBody || ctx.destResBody.length !== ctx.destRes.headers['content-length'];
 }
 
-function callResponseEventCallbackForProcessedRequest (ctx, rule) {
-    const responseInfo           = requestEventInfo.createResponseInfo(ctx);
-    const configureResponseEvent = new ConfigureResponseEvent(rule, ConfigureResponseEventOptions.DEFAULT);
-
-    ctx.session.callRequestEventCallback(REQUEST_EVENT_NAMES.onConfigureResponse, rule, configureResponseEvent);
-
+function callResponseEventCallbackForProcessedRequest (ctx, configureResponseEvent) {
+    const responseInfo         = requestEventInfo.createResponseInfo(ctx);
     const preparedResponseInfo = requestEventInfo.prepareEventData(responseInfo, configureResponseEvent.opts);
-    const responseEvent        = new ResponseEvent(rule, preparedResponseInfo);
+    const responseEvent        = new ResponseEvent(configureResponseEvent._requestFilterRule, preparedResponseInfo);
```

Comment:
@miherlosev 
i've refactored in order to split this function.

I really don't like passing the `configureResponseEvent._requestFilterRule,` because `_requestFilterRule` is like a _private_ property.

What is it preferable?:
1. Leave it like this.
2. This function needs the `configureResponseEvent` object because, in order to call `requestEventInfo.prepareEventData`, i need `configureResponseEvent.opts`. This dependency could be removed as this `opts` are always `ConfigureResponseEventOptions.DEFAULT`. The `callResponseEventCallbackForProcessedRequest` interface could be `callResponseEventCallbackForProcessedRequest (ctx, rule)`, in `sendProxyResponse`, i would iterate through the rules array (instead of the `configureResponseEvents`) and here i would directly use ``ConfigureResponseEventOptions.DEFAULT`` (insead of `configureResponseEvent.opts`).
3. In [ConfigureResponseEvent](https://github.com/DevExpress/testcafe-hammerhead/blob/0dcf519e75e21b6535708df53fac0f8a5e5cc6fb/src/session/events/configure-response-event.js) class rename the `_requestFilterRule` property to `requestFilterRule`.
4. In [ConfigureResponseEvent](https://github.com/DevExpress/testcafe-hammerhead/blob/0dcf519e75e21b6535708df53fac0f8a5e5cc6fb/src/session/events/configure-response-event.js) class add a getter: `get requestFilterRule() { return this._requestFilterRule; }`

Reply:
I suggest you leave it as is, because we will keep the `_requestFilterRule` property as a private field. 
In the future, we will mark the `_onConfigreResponse` event as public. This is why, the `_requestFilterRule` property should not be exposed as a public property of the `ConfigureResponseEvent`.